src/component/container/page_list.tsx

```diff
@@ -11,14 +11,109 @@ export interface PageListProps {
 	store: Store;
 }
 
+export interface PageListItemProps {
+	id: string;
+	name: string;
+	pageRef: PageRef;
+	store: Store;
+}
+
+@observer
+export class PageListItem extends React.Component<PageListItemProps> {
+	@MobX.observable protected pageElementEditable: boolean = false;
+	@MobX.observable protected pageName: string;
+
+	public constructor(props: PageListItemProps) {
+		super(props);
+
+		this.handleInputChange = this.handleInputChange.bind(this);
+		this.handleBlur = this.handleBlur.bind(this);
+		this.handlePageKeyDown = this.handlePageKeyDown.bind(this);
+		this.handlePageClick = this.handlePageClick.bind(this);
+		this.handlePageDoubleClick = this.handlePageDoubleClick.bind(this);
+		this.renamePage = this.renamePage.bind(this);
+	}
+	public render(): JSX.Element {
+		return (
+			<DropdownItemEditableLink
+				editable={this.pageElementEditable}
+				focused={this.pageElementEditable}
+				handleChange={this.handleInputChange}
+				handleClick={(e: React.MouseEvent<HTMLElement>) => {
+					e.preventDefault();
+					this.handlePageClick(this.props.id);
+				}}
+				handleDoubleClick={this.handlePageDoubleClick}
+				handleKeyDown={this.handlePageKeyDown}
+				name={this.props.name}
+				handleBlur={this.handleBlur}
+				placeholder={this.props.name}
+				value={this.pageName ? this.pageName : this.props.name}
+			/>
+		);
+	}
+
+	protected handlePageClick(id: string): void {
+		this.props.store.openPage(id);
+	}
+
+	@MobX.action
+	protected handleBlur(): void {
+		this.pageElementEditable = false;
+	}
+
+	@MobX.action
+	protected handlePageDoubleClick(): void {
+		this.pageElementEditable = !this.pageElementEditable;
+	}
+
+	@MobX.action
+	protected handlePageKeyDown(e: React.KeyboardEvent<HTMLInputElement>): void {
+		switch (e.key.toString()) {
+			case 'Escape':
+				this.pageElementEditable = false;
```

Comment:
You should clear `this.pageName`. If you don't do this it will be auto filled for the next edit process

Reply:
This should happen whenever you set `pageElementEditable` to false

Reply:
I guess @tilmx expectation is not to completely clear the name. But what I can do is set it to the last available name. What do you think?

Reply:
The problem is with this it feels really strange.
I open the edit mode and write something in. I then press escape because i wan't to write something completely different and when i then go back in to the field i have the value from before.

Reply:
Example:
![rename-page-example](https://user-images.githubusercontent.com/2543933/36031649-c4e7edc2-0dab-11e8-9b6f-3a676078ff9b.gif)

Reply:
Hi @LKuechler,

that is the expect UX behavior, please check my example:

![kapture 2018-02-12 at 10 43 38](https://user-images.githubusercontent.com/3357315/36090702-aa620d56-0fe1-11e8-9740-87cf2a95d5bd.gif)

Reply:
@Palmaswell your gif shows a completely different use case of the ui.
Please follow these steps:
1. Edit a page name
2. Write something in it
3. Cancel the edit process (esc or click next to the fiel)
4. Now open the edit dialog again

Reply:
Step 4: I could clear to the last saved page name. What do you think about that @LKuechler @tilmx?

Reply:
@Palmaswell 
I don't think that the last saved page name is needed. If `this.pageNameInputValue` is really cleared the `this.props.name` is used as fallback and with that we have the logic that you and @tilmx want. Without having the bug, that unused values are carried in to the next edit.

Reply:
@LKuechler done ✅